Adding a detailed overview diagram to appendix

shigeya opened the pull request:
**Experimental**
Keeping old figure for review.

Related to #453.  Trying to include all of the relationships of the entities described in 1.3. Architecture Overview.

This diagram does not display DID methods, DID resolvers, and DID URL dereferences in this picture since their position in this level of abstraction is unclear.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/shigeya/did-core/pull/600.html" title="Last updated on Feb 10, 2021, 4:34 AM UTC (36b8395)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/600/59687a4...shigeya:36b8395.html" title="Last updated on Feb 10, 2021, 4:34 AM UTC (36b8395)">Diff</a>

shigeya commented:
Note that we need consistency of notation/expression/sizes/whatever among the figures to be consistent.
I believe I can update figures as part of editorial fixes. So, I let the color black and white for now.

shigeya commented:
> This diagram does not display DID methods, DID resolvers, and DID URL dereferences in this picture since their position in this level of abstraction is unclear.

I tried to place them into the figure. Texts are smaller than I want, but now every component described in the section is in the diagram.

shigeya commented:
The preview link above doesn't work. For the review, I added the following PNG here.

![did_architecture_overview-2](https://user-images.githubusercontent.com/69672/106943386-ba9c5780-6768-11eb-9179-e37b919dff43.png)

shigeya commented:
Minor fix

![did_architecture_overview-2](https://user-images.githubusercontent.com/69672/106944368-d5bb9700-6769-11eb-9f45-a6bce466720a.png)

shigeya commented:
Minor update to add self-reference dereference to DID Document.
![did_architecture_overview-2](https://user-images.githubusercontent.com/69672/106972051-72def580-6793-11eb-880b-1b65b9616048.png)

shigeya commented:
PR includes the SVG file for the above figure. This time, I also committed the corresponding PNG file too.

OR13 reviewed (approved):
This diagrams is one of the greatest contributions I am seen... massively helpful.

shigeya commented:
Clean-up of the PR. Also fixed the text `DID document-local` to `DID document-relative.`
![did_architecture_overview-2](https://user-images.githubusercontent.com/69672/107187632-a5a21b80-6a29-11eb-9527-08867d5f4813.png)

shigeya commented:
Two issues on the figure
1. It's correctly drawn with Safari but not with Firefox(85.0) and Chrome(88), as attached image (properties inside the DID document box): <img width="820" alt="ScreenShot 2021-02-08 17 18 45" src="https://user-images.githubusercontent.com/69672/107193331-11888200-6a32-11eb-8392-99a73e026ace.png">
2. Probably need to draw a relationship between DID resolver and DID dereferencer.

shigeya commented:
Fixed the compatibility issue mentioned above.

TallTed commented on diagrams/did_architecture_overview-2.svg:
```diff
@@ -0,0 +1,400 @@
+      <g id="Graphic_64">
+        <text transform="translate(175.07875 305.47245)" fill="black">
+          <tspan font-family="Helvetica Neue" font-size="12" font-weight="400" fill="black" x="7531753e-19" y="11">fragment refers, and</tspan>
+          <tspan font-family="Helvetica Neue" font-size="12" font-style="italic" font-weight="700" fill="black" x="19.332" y="26.336">dereferences</tspan>
```
```suggestion
          <tspan font-family="Helvetica Neue" font-size="12" font-style="italic" font-weight="700" fill="black" x="19.332" y="26.336">dereferences,</tspan>
```

shigeya replied:
Thanks for the suggestion. fixed.

msporny reviewed (approved):
While I agree that the diagram is accurate -- it exposes the reader to a large number of details up front and makes the DID ecosystem look horribly complicated.

I would prefer to have the diagram in an appendix -- we went to great lengths in the Verifiable Credentials specification to simplify the diagram in the overview because we knew people would cut/paste it into their presentations. That bet paid off in a big way.

While this diagram is accurate, I'm very concerned about people copying it into their presentations -- it'll make the specification and technology come off as a complex, academic, and inaccessible.

I'm approving the diagram because I do think it's very useful -- but in an appendix, after people have read through the entire spec (or the more complicated portions of it). I think we need a summary diagram in the overview section... and the overview section needs to be simplified.

shigeya commented:
@msporny I understand your concern and I fully agree with you. Let me try to create a simpler diagram based on this in 13 hours (It's about 11 am in AoE clock: https://time.is/Anywhere_on_Earth).

shigeya commented:
A brief version of the above is created and sent as another PR #612. I will update this PR to relocate the diagram to the appendix section.

shigeya commented:
Relocated to the appendix, prior to FAQ. Attached the image for the convenience of the review.

To make use of the rich real estate of the appendix section, adjusted the diagram to be narrower, taller. Adjusted to use larger font size and little extra spaces between texts, boxes, and lines.

Plus, some minor clean-ups including "did:" to be text, not in a box.
![did_detailed_architecture_overview](https://user-images.githubusercontent.com/69672/107464546-e887fe80-6ba3-11eb-874a-78c0420a3f70.png)

msporny commented:
Editorial, multiple reviews, changes requested and made, no objections, merging.
